ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -759,6 +759,21 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 		deleted := struct{}{}
 		numDeletedUnhealthyWorkerPods := 0
 		for _, workerPod := range workerPods.Items {
+			if workerPod.ObjectMeta.DeletionTimestamp != nil {
```

Comment:
I'm not sure how much this check helps. 

Even with a finalizer configured, if a pod has been deleted with a deletion timestamp it is already terminating. We will keep the Pod object around but the worker will not actually be available.

Reply:
I think we are trying to avoid admission webhooks as much as possible. @kevin85421 thoughts?